src/core/device/cuda_gpu.cc

```diff
@@ -120,9 +132,41 @@ void CudaGPU::TimeProfilingDoExec(function<void(Context*)>&& fn, int executor,
   cudaEventCreate(&(node->start_));
   cudaEventCreate(&(node->end_));
 
+#ifdef USE_DIST
```

Comment:
Do we have a better scheme?

Reply:
If I use USE_CUDA instead if USE_DIST in common.h, this @ifdef can be deleted

Reply:
which one do you prefer? I can use USE_CUDA in common.h instead

Reply:
I mean not to record the event based on the operator name.

Reply:
then may pass the cuda stream from the buffered op and save the cuda stream in the node?

Reply:
Yeah, we need big changes to solve this problem(e.g. passing a struct named OpDescriptor to the device). I think we just keep it like this for now.

Reply:
> Yeah, we need big changes to solve this problem(e.g. passing a struct named OpDescriptor to the device). I think we just keep it like this for now.

Yes, I guess OpDescriptor would be useful to record many context regarding the operator. I think it is good to include it in the future